### PR TITLE
modify consul demo to support makefile

### PR DESCRIPTION
**Pull request by xiaoliu10**
1.  support makefile
2.  remove dead code

### CODE REVIEW

**Review by zouyx — COMMENTED** [review]
You should add sample into https://github.com/apache/dubbo-go-samples/blob/master/start_integrate_test.sh for integration test.

**Review by zouyx — COMMENTED** [review]
Action not pass

File: registry/servicediscovery/consul/go-client/cmd/client.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ package main
 import (
 	"context"
 	"fmt"
+	hessian "github.com/apache/dubbo-go-hessian2"
```
Comment (AlexStocks) [inline registry/servicediscovery/consul/go-client/cmd/client.go:23]: hey, guy, move it, pls.

File: registry/servicediscovery/consul/go-client/cmd/client.go [hunk 2]
```diff
@@ -20,6 +20,8 @@ package main
...
+	"github.com/apache/dubbo-go-samples/registry/servicediscovery/consul/go-client/pkg"
```
Comment (AlexStocks) [inline registry/servicediscovery/consul/go-client/cmd/client.go:24]: the same as above.

File: registry/servicediscovery/consul/go-client/pkg/user.go [hunk 3]
```diff
@@ -15,25 +15,13 @@
  * limitations under the License.
  */
 
-package main
+package pkg
```
Comment (AlexStocks) [inline registry/servicediscovery/consul/go-client/pkg/user.go:18]: maybe u can change the dir name. 'pkg' is not so good.

**Review by AlexStocks — APPROVED** [review]

File: registry/servicediscovery/consul/go-client/pkg/user.go (hunk 3 above)
Comment (xiaoliu10) [reply registry/servicediscovery/consul/go-client/pkg/user.go:18]: all samples use pkg ,  so do i

File: registry/servicediscovery/consul/go-client/cmd/client.go [hunk 4]
```diff
@@ -27,6 +27,11 @@ import (
 )
 
 import (
+	"github.com/apache/dubbo-go-samples/registry/servicediscovery/consul/go-client/pkg"
```
Comment (AlexStocks) [inline registry/servicediscovery/consul/go-client/cmd/client.go:30]: the position of the package itself should be the third import block.

File: registry/servicediscovery/consul/go-server/tests/integration/main_test.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	_ "github.com/apache/dubbo-go/metadata/report/consul"
+	_ "github.com/apache/dubbo-go/metadata/service/remote"
+	_ "github.com/apache/dubbo-go/protocol/dubbo"
+	_ "github.com/apache/dubbo-go/protocol/jsonrpc"
+	_ "github.com/apache/dubbo-go/registry/consul"
+	_ "github.com/apache/dubbo-go/registry/protocol"
+	_ "github.com/apache/dubbo-go/registry/servicediscovery"
+)
+
+import (
```
Comment (AlexStocks) [inline registry/servicediscovery/consul/go-server/tests/integration/main_test.go:40]: move it to 1st import block.

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**